pkg/store/bucket.go

```diff
@@ -2299,6 +2305,9 @@ func toPostingGroup(lvalsFn func(name string) ([]string, error), m *labels.Match
 
 		var toRemove []labels.Label
 		for _, val := range vals {
+			if ctx.Err() != nil {
```

Comment:
This could be a double edged sword because `vals` is often a high number, so we would end up checking for cancellation thousands of times.

Reply:
That's correct. Do you have any suggestions on where we should check context?

Reply:
yeah... =/

I did not find a better way as well to do on the prometheus PR. Maybe we should check the ctx.done channel somehow?

Reply:
I think we would still need a mutex for the `cancelled` variable, which would be the same as checking `ctx.Err()`. Is there any downside to just checking the context before the `vals` loop?

Reply:
I see, we can try this and see if it ends up being a problem.